sphinx/source/docs/user_guide/bokehjs.rst

```diff
@@ -280,6 +280,93 @@ and hover policy. Here is an example of a ``bar`` chart and the plot it generate
 
     plt.show(plt.gridplot([[p1, p2], [p3, p4]], {plot_width:350, plot_height:350}));
 
+.. _userguide_bokehjs_differences:
+
+Usage: Differences to Python library
+------------------------------------
+
+While the BokehJS models generally match the Python equivalents, there are some cases
+where usage can differ significantly between BokehJS and the Python library. Some
+instances we have identified are listed below.
+
+.. _userguide_bokehjs_differences_titles:
+
+Updating figure titles
+~~~~~~~~~~~~~~~~~~~~~~
+
+Calling ``var plot = new Bokeh.Plotting.figure({title:"some title"})`` replaces
+plot.title (a ``Bokeh.Title()`` object by default) with the provided string.
+This makes it so the title cannot be dynamically updated later. In Python, the
+equivalent approach casts the string to the underlying Title() object.
+
+To preserve the ability to update the title later, use the following sequence instead:
+
+.. code-block:: javascript
+
+    var plot = new Bokeh.Plotting.figure()
+    plot.title.text = "some title"
+
+.. _userguide_bokehjs_differences_gridplots:
+
+Using ``sizing_mode`` in gridplot layouts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``Bokeh.Plotting.gridplot()`` does not inherit ``sizing_mode`` from the underlying plots.
+To match the expected behavior from the Python equivalent, for example when using
+``sizing_mode: 'stretch_width'``, the ``sizing_mode`` must be explicitly stated for both
+the ``figure()`` and the ``gridplot()`` objects.
+
+.. _userguide_bokehjs_differences_range_padding:
+
+Making axis range properties persitent
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Setting ``plot.x_range.start`` and ``plot.x_range.end``, or ``plot.x_range.range_padding``
+to adjust whitespace around glyphs on an X axis (or the corresponding properties for a
+Y axis) is not persistent across resets. In the Python library, the values for these
+properties are persistent when a figure is reset. To achieve similar behavior in BokehJS,
+set both ``plot.x_range.range_padding`` and ``plot.x_range._initial_range_padding``.
+
+.. _userguide_bokehjs_differences_palette:
+
+Palettes for ``Bokeh.Plotting`` objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The color palette is available under ``Bokeh.Charts`` but not under ``Bokeh.Plotting``.
+To use it with ``Bokeh.Plotting`` objects such as ``Bokeh.Plotting.image()``, the
+palette argument must be provided as a color string array, rather than as a string.
+
+.. _userguide_bokehjs_differences_name_field:
+
+Setting glyph names for ``HoverTool``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Calling ``plot.line([0,1],[0,1],{name:“the_name”})`` sets the name under ``renderer.glyph.name``,
+rather than ``renderer.name``. The latter is used for HoverTool, under the special field
+``$name``. To enable this special field in the HoverTool, pass the glyph name to the renderer:
+
+.. code-block:: javascript
+
+    plot.renderers[0].name = plot.renderers[0].glyph.name;
```

Comment:
This is fixed in PR #11018.

Reply:
@mattpap Awesome! That will be nice to have built in. For now tho, since #11018 is pointing to branch-2.4, I assume we'd still want an update to the live 2.3 docs. That is, assuming @bryevdv's position on redeploying docs when this PR is ready still stands?

Reply:
To be honest @oviquezr I am having some second thoughts about this as it is currently formulated. What I would actually personally like to limit a "Differences between Python and JS" section to is *intentional* differences, not bugs. The best place to document bugs is in the issue tracker (not least because they will accurately represent status by being open or closed)

I think what I might prefer for this is to:

* Make individual issues on GitHub for all these bugs (including any workarounds)
* Link the *issues* in the tracker under a section called "Known issues"

That would:

* reduce the docs upkeep burden (the issues will always be up to date, necessarily)
* reduce duplication between the tracker and docs
* leave a future "differnces" section open to be clearly for describing intentional differences

I would be *much* more comfortable with that. 

cc @tcmetzger

Reply:
FYI there is a Bokeh sphinx extension for linking issues:
```
:bokeh-issue:`1234`
```